06week/higherOrder.js

```diff
@@ -2,38 +2,68 @@
 
 const assert = require('assert');
 
-function forEach(arr, callback) {
-  // Your code here
-}
-
+// function forEach(arr, callback) {
+//   // Your code here
+// }
+const arr = [10, 20, 30];
 function map(arr, callback) {
```

Comment:
Here we're declaring a function called `map()` . We should have a function declraration exactly liek this for both filter and reduce looking like this:

`function filter(array, callback, thisObject) {
    // filter function code here
}`

`function reduce(array, accumulator) {
    // reduce function code here
}`

Then when we call filter and reduce, we do it like this:

```
const reduceIt = reduce( [1, 2, 3, 4] );
console.log(reduceIt);
```

```
const filterIt = filter( [ 10, 20, 60, 90, 100 ], function(i) {
    return i <= 60; // We are only returning the numbers equal or less than 60 from the original array
});
console.log(filterIt); // Output: [ 10, 20, 60 ]
```